src/main/java/io/vertx/core/http/impl/HttpServerResponseImpl.java

```diff
@@ -367,20 +368,23 @@ public void end(Buffer chunk, Handler<AsyncResult<Void>> handler) {
   }
 
   private void end(Buffer chunk, ChannelPromise promise) {
+    end(chunk.getByteBuf(), promise);
+  }
+
+  private void end(ByteBuf chunk, ChannelPromise promise) {
```

Comment:
can we avoid adding a new method ? that decrease the chances of inlining

Reply:
This method should be trivially inlineable IIRC so it shouldn't reduce the inlining budget IIRC: I'll check better to be sure and will create a new whole method if needed

Reply:
I'm still checking here http://hg.openjdk.java.net/jdk8/jdk8/hotspot/file/87ee5ee27509/src/share/vm/opto/bytecodeInfo.cpp#l331

Reply:
After the last commits on this PR I can say that the inlining budget secret with trivially inlineable callee too :) good suggestion!